src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -113,7 +113,16 @@ private int getFromValueIfSingleShard(final NormalizationProcessorWorkflowExecut
         if (searchPhaseContext.getNumShards() > 1 || request.fetchSearchResultOptional.isEmpty()) {
             return -1;
         }
-        return searchPhaseContext.getRequest().source().from();
+        int from = searchPhaseContext.getRequest().source().from();
+        // for the initial searchRequest, it creates a default search context which sets the value of
+        // from to 0 if it's -1. That's not the case with SearchPhaseContext, that's why need to
+        // explicitly set to 0 for the single shard case
+        // Ref:
+        // https://github.com/opensearch-project/OpenSearch/blob/2.18/server/src/main/java/org/opensearch/search/DefaultSearchContext.java#L288
```

Comment:
I see that not only from value got it's default, but also the size. Should we care about size too?

Reply:
Changing size might affect the fetch phase and would need more deep dive. So, let's keep this change for `from`